File: fix_paths.js
const fs = require('fs');
const path = require('path');

const configPath = path.join(__dirname, 'config.json');
const config = JSON.parse(fs.readFileSync(configPath, 'utf-8'));

const base = 'https://abu168888.github.io/tvbox-config/';

// Fix spider path
if (config.spider && config.spider.startsWith('./')) {
    config.spider = base + config.spider.substring(2);
}

// Fix lives
for (const live of config.lives || []) {
    if (live.url && live.url.startsWith('./')) {
        live.url = base + live.url.substring(2);
    }
}

// Fix site ext paths
for (const site of config.sites || []) {
    if (typeof site.ext === 'string' && site.ext.startsWith('./')) {
        site.ext = base + site.ext.substring(2);
    }
    if (typeof site.ext === 'object' && site.ext !== null) {
        for (const [k, v] of Object.entries(site.ext)) {
            if (typeof v === 'string' && v.startsWith('./')) {
                site.ext[k] = base + v.substring(2);
            }
        }
    }
}

// Write with BOM for UTF-8 compatibility
const utf8BOM = '\uFEFF';
fs.writeFileSync(configPath, utf8BOM + JSON.stringify(config, null, 2), 'utf-8');

console.log('Config updated with GitHub Pages URLs');
console.log('Spider:', config.spider);
console.log('Sites count:', config.sites?.length);
console.log('First site ext:', config.sites?.[0]?.ext);

File: fix_paths.py
import json
import sys

config_path = 'config.json'
base_url = 'https://abu168888.github.io/tvbox-config/'

with open(config_path, 'r', encoding='utf-8') as f:
    config = json.load(f)

print(f"Original spider: {config['spider']}")

# Fix spider path
if config['spider'].startswith('./'):
    config['spider'] = base_url + config['spider'][2:]

# Fix lives
for live in config.get('lives', []):
    if live.get('url') and live['url'].startswith('./'):
        live['url'] = base_url + live['url'][2:]

# Fix site ext paths
count = 0
for site in config.get('sites', []):
    if isinstance(site.get('ext'), str) and site['ext'].startswith('./'):
        site['ext'] = base_url + site['ext'][2:]
        count += 1
    elif isinstance(site.get('ext'), dict):
        for key in site['ext']:
            if isinstance(site['ext'][key], str) and site['ext'][key].startswith('./'):
                site['ext'][key] = base_url + site['ext'][key][2:]
                count += 1

print(f"Fixed spider: {config['spider']}")
print(f"Fixed {count} site ext paths")

with open(config_path, 'w', encoding='utf-8') as f:
    json.dump(config, f, indent=2, ensure_ascii=False)

print("✅ Config file updated successfully!")

# Verify
with open(config_path, 'r', encoding='utf-8') as f:
    verify = json.load(f)
    
print("\nVerification - First 3 sites:")
print(json.dumps(verify['sites'][:3], indent=2, ensure_ascii=False))
